test/acceptance_tests/acceptancetest_publish_test.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include <ros/ros.h>
+#include <sensor_msgs/LaserScan.h>
+
+int unsigned constexpr TIMEOUT_LENGTH_S = 100;
+
+bool message_received = false;
+bool timeout_reached = false;
+
+void callback(const sensor_msgs::LaserScan::ConstPtr& msg)
+{
+  message_received = true;
+}
+
+int main(int argc, char** argv)
```

Comment:
What about http://wiki.ros.org/rostest/Nodes#publishtest ?

Reply:
This test is not supposed to be executed by travis, thats why its a separate node, that is added as executable not as test in the cmake.